### PR TITLE
`General`: Improve exercise view when using LTI

### PR DESCRIPTION
**Pull request by raffifasaro**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data (e.g. using paging).
- [x] I **strictly** followed the principle of **data economy** for all client-server REST calls.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [ ] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

The LTI iFrame was displaying footer, header and sidebars. Since moodle is already used to navigate the exercises we can omit most of these to get a more focused and cleaner look.
image

### Description

image

The changes mainly concern the creation of a new Lti frontend service to track if the frontend is launched by LTI or not. This is used to remove unwanted elements from different components.

### Steps for Testing

Prerequisites:
- Access to Moodle
- Testserver 1

1. Make sure you are logged in to Artemis with a test user (artemis_test_user_{1-5})
2. Navigate to [Moodle](https://moodle.ase.in.tum.de) and login with the same test user (artemis_test_user_{1-5}, same PW as for the Artemis Test Server)
3. Go to My Courses and open the course TS1 - Artemis Feature Demo Course
4. Click on one of the exercise
5. Artemis should open in an iFrame and display just the main exercise body as shown in the screenshots.
(If Moodle throws an error in the iFrame, just refresh the page)

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Performance Review

- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance even for very large courses with more than 2000 students.
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance even for very large courses with more than 2000 students.
#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2

#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| lti13-exercise-launch.component.ts | 90% | ✅ |
| course-exercises.component.ts | 89.7% | ✅ |
| course-overview.component.ts | 86.39% | ✅ |
| main.component.ts | 73.61% | ✅ |
| lti.service.ts | 83.33% | ✅ |

### Screenshots

image
image

## Summary by CodeRabbit

- **New Features**
	- Introduced LTI support with conditional rendering based on LTI context.
	- Added a new `LtiService` to manage LTI state across components.
	- Enhanced component responsiveness with conditional styling based on LTI state.
- **Bug Fixes**
	- Improved error handling in the `onResize` method to prevent potential issues.
- **Tests**
	- Enhanced test suite for `CourseOverviewComponent` with improved assertions and logic checks.

### CODE REVIEW

**Review by az108 — DISMISSED** [review]
Tested on TS1 as described and works as expected. The exercise is loaded as in the provided screenshots:

image

**Review by az108 — CHANGES_REQUESTED** [review]
after the recent changes, I sadly get following message and not the exercise anymore

image

**Comment by raffifasaro** [comment]
> after the recent changes, I sadly get following message and not the exercise anymore
> 
> image

This is a common issue with Moodle/LTI, refreshing the page should fix it.

**Review by az108 — APPROVED** [review]
Thanks for clarifying, still works 👍

**Review by pzdr7 — APPROVED** [review]
Tested on TS1.

The exercise loads and the sidebar, header, and footer are not shown.

image

While loading, the sidebar pops up for a moment. Maybe in a follow-up, you can reduce this delay.

image

File: src/main/webapp/app/overview/course-overview.component.ts [hunk 1]
```diff
@@ -234,13 +238,18 @@ export class CourseOverviewComponent implements OnInit, OnDestroy, AfterViewInit
...
+        });
     }
 
     /** Listen window resize event by height */
     @HostListener('window: resize', ['$event'])
     onResize() {
-        this.updateVisibleNavbarItems(window.innerHeight);
-        if (!this.anyItemHidden) this.itemsDrop.close();
+        if (this.itemsDrop) {
+            this.updateVisibleNavbarItems(window.innerHeight);
```
Comment (iyannsch) [inline src/main/webapp/app/overview/course-overview.component.ts:250]: Is the `updateNavbarItems` functionality really strictly bound to the `itemsDrop`? I'm not 100% certain with that functionality, thus, the question.. :)

**Review by iyannsch — CHANGES_REQUESTED** [review]
Code looks good, thanks for the work! I've got 1 small question inline and would love some clarity there ;)

**Review by krusche — APPROVED** [review]
The changes look good to me 👍 We can implement additional improvements in follow-up PRs

**Pull request merged**